java/Nikiforov-Oleksandr/src/com/zoolatech/lecture7/tasks/_2/Calculator.java

```diff
@@ -0,0 +1,21 @@
+package com.zoolatech.lecture7.tasks._2;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public abstract class Calculator {
+    protected int value;
+
+    public Calculator(int value) {
+        this.value = value;
+    }
+
+    public void add(int valueToAdd) {
+        value += valueToAdd;
+    }
+
+    public void subtract(int valueToSubtract) {
+        value -= valueToSubtract;
+    }
+
+    public abstract <T> T getValue();
```

Comment:
other methods of this class work with int values, and this method is generic - you need to make all methods to work with int values, or make all methods generic